plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -232,7 +232,8 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, PreparedQuery pr
         try (Connection connection = connectionFactory.openConnection(session);
                 PreparedStatement preparedStatement = queryBuilder.prepareStatement(this, session, connection, preparedQuery)) {
             ResultSetMetaData metadata = preparedStatement.getMetaData();
-            if (metadata == null) {
+            // SingleStore returns empty metadata instead of a null value
+            if (metadata == null || metadata.getColumnCount() == 0) {
```

Comment:
I'm not sure if this change is acceptable or not since existing MariaDB failure case depends on this block. 
(That is why I was going to fix the test message in #12696)

cc: @kasiafi @findepi

Reply:
As I've stated in a commit message: constructing Query PTF without columns doesn't make sense any way so this condition seems fine to me.

Reply:
This method is supposed to create a table handle for a PreparedQuery. Apparently, it is able to create a valid TableHandle with no columns. Let the caller take responsibility of handling the "no columns" condition. A TableHandle without columns might make no sense for a Table Function, but it might be OK in some other context (in the future). If a TableHandle with no columns is never OK, then maybe we miss a check in the JdbcTableHandle constructor?

Reply:
A relation should have at least one column (though there exist sub-plans with no columns as a result of optimization).

My point is that a TH with no columns could still be potentially useful if it represents e.g. an insert. Adding the column on Trino side would not be a big issue.